### PR TITLE
[WIP][INFRA-1657] Build & Release Jenkins

### PR DESCRIPTION
**Pull request by olblak**

### CODE REVIEW

**Comment by daniel-beck** [comment]
This mentions the security releases, but does not mention Maven staging repositories, or selecting a different source code repository (like `jenkinsci-cert/jenkins`), or pushing commits and tag manually after staging -- in fact, the Maven commands listed are not suitable to stage releases.

(I expect that would still be included anyway, this is WIP after all, but just to make sure it doesn't fall off the table 😃 )

**Comment by slide** [comment]
One item that also needs to be taken into account in the Credentials area is a signing certificate for Windows installers. signtool (the MS tool for signing) requires the certificate to be in the following format: 

"A valid code signing certificate, for example, a Personal Information Exchange (.pfx) file created with the MakeCert.exe and Pvk2Pfx.exe tools"

References for signing MSI installers can be found here: 

https://docs.microsoft.com/en-us/windows/desktop/appxpkg/how-to-sign-a-package-using-signtool

File: iep-012/README.adoc [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+|===
+
+=== Workflow
+It's now time to cover the last topic, which is how we build, test and release a specific version for an environment. Jenkins is used as the conductor by means of Jenkinsfile and will execute a succession of actions defined link:https://github.com/olblak/jenkins/blob/master/Jenkinsfile.release[here].
+In order to simplify the maintainability of this script, most of the logic as been moved to a shell link:https://github.com/olblak/jenkins/blob/master/scripts/buildJenkins.bash[script] that accept various arguments.
+
+All endpoints are hard-coded with Kubernetes internal endpoint like "repo", jenkins, ... And any other configuration is done via environment variables configured at the environment level.
+So every environment use the same Jenkinsfile but the output change depending on the environment configuration and the git repository.
+
+Maven with the link:http://maven.apache.org/maven-release/maven-release-plugin/[release plugin] is used to build Jenkins(link:https://wiki.jenkins.io/pages/viewpage.action?pageId=3309681[why?]).
```
Comment (jglick) [inline iep-012/README.adoc:124]: Well that page is about using Maven generally, not `maven-release-plugin` specifically.

**Comment by olblak** [comment]
I am going to merge this PR and open a new one later one with further improvements

**Pull request merged**